packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -385,17 +388,8 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     element: ReactElement,
     expirationTime: ExpirationTime,
   ): Fiber {
-    if (current === null || current.type !== element.type) {
-      // Insert
-      const created = createFiberFromElement(
-        element,
-        returnFiber.internalContextTag,
-        expirationTime,
-      );
-      created.ref = coerceRef(current, element);
-      created.return = returnFiber;
-      return created;
-    } else {
+    // TODO: Split these into branches based on typeof type
```

Comment:
https://github.com/facebook/react/issues/11290 It's to ensure correctness between type checking when comparing across functions, classes, strings, and potentially numbers/symbols.

Reply:
This should split based on `tag`, not `typeof`. It doesn't really matter if you're going to check the type anyway. Some of the other duplicates of this comment is confusing since in those cases you already do that branching.

Reply:
In the duplicates of this comment, I only do branching on `tag === REACT_FRAGMENT_TYPE`. My understanding was that we wanted to branch in such a way so that we only compare types of the same type (`function`, `string`, `number`, `symbol`, etc.). Perhaps I misunderstood.